### ISSUE TITLE
Multi-selected blocks + contentOnly representations: always show text styles if range includes text

### ISSUE THREAD

**Issue by jasmussen** [WordPress/gutenberg#65645]
If you select two paragraphs, you get text formatting tools across, in the inspector:

![two text blocks](https://github.com/user-attachments/assets/c42fa1cd-dd19-4dae-a013-85e2f6a3dae6)

If you select two image blocks, you get image formatting tools across, in the inspector:

![two images](https://github.com/user-attachments/assets/05681858-3d78-4b32-9603-a87b5b5fa74a)

If you select mixed content, such as a paragraph and an image, you get no controls at all, in the inspector:

![paragraph and image](https://github.com/user-attachments/assets/21cc23d9-3533-4518-a677-eda5445dbccb)

This is not very useful.

## Proposed solution

If you have a mixed selection that includes text, always show the text formatting tools.

![paragraph-and-image-with-formatting-tools](https://github.com/user-attachments/assets/3dbbd1d9-9033-407e-9674-12f2063714ad)

**Comment by Mamaduka** [comment]
IIRC, this is intentional. The inspector controls are only displayed when blocks of the same type are selected since they share attribute definitions.

Side note: Multi-editing blocks have their issues when specific changes might get lost. See #51609 and #41260.

**Comment by jasmussen** [comment]
Yes, it definitely feels intentional so I'm not suggesting this as a bug, or even an urgent one, can be 6.8. But mainly, it feels like whenever you have text blocks as part of your selection, surfacing text-block attributes across is worth it. Kind of like how you might select across text and images in a Google Doc and change press ⌘B, you'd at least make the text bold even if not the image. So it's a writing flow suggestion with lowish priority.

**Comment by jasmussen** [comment]
Updated this issue to include that these tools should also be shown in contentOnly representations!

**Comment by jasmussen** [comment]
Noting that I updated the title of this post to be clearer. Having this work in contentOnly makes sense too, but that's really secondary to the primary use case, simply allowing more controls in the inspector at all times. 

In that vein, I also recently revisited these mockups. Consider the case of selecting multiple paragraphs, today you _do_ see a useful inspector, showing the repeated panels across all. However a shortcoming is that **even if the various paragraphs selected have mixed values, e.g. different colours or font sizes, the values from the first block are shown**. It's still useful, because you can quickly harmonise them in a few clicks, but worth pointing out. An improvement here would be to show the value "Mixed", where we can.

For the use case of selecting a paragraph, an image, and another paragraph with different values, here's a fresh mockup:

<img width="1360" height="1090" alt="Image" src="https://github.com/user-attachments/assets/5e153be8-7b60-499f-a762-494d200528eb" />

* Note how the colour swatch shows a "Mixed" value.
* Note how the Borders panel remains available, because it works across both paragraphs and images.

I also included "Resolution" settings, because I selfishly want to see that across multiple images selected. Whether that's a fit or not, up to a developer!

---

I also made a slightly crazier mockup. I kept thinking about: what should we show in the inspector when you have selected mixed content? What if you select a container block with a lot of child blocks inside, should we show values from the children? The mockup below did not come together super well, I consider it incomplete. But I'm sharing it nevertheless, because I think there's an element of an idea here, that perhaps in someone else's hands, can become a complete idea:

<img width="1360" height="1804" alt="Image" src="https://github.com/user-attachments/assets/035402a5-f773-43e0-b9a2-7ece4e97f0ed" />

E.g. @fcoveram mentioned that there may be something to combining the Elements and proposed Selection panels into one: it can remain collaposed by default, and it can possibly surface local overrides and be verbose. But mostly, I like the idea that it can show all sorts of properties of child blocks, giving a power-user tool to quickly override them. If we get this right, this panel is conceptually the same as what we talked about building for contentOnly:

<img width="280" height="896" alt="Image" src="https://github.com/user-attachments/assets/7e25f6b6-d3eb-4f1c-8de7-a8de2460357a" />

**Comment by jasmussen** [comment]
CC: @WordPress/gutenberg-design

**Comment by fcoveram** [comment]
I like the idea of showing controls for the elements that are part of the selection, even if some of them don't impact some blocks selected directly, like image resolution when you have paragraph blocks in the selection. Showing something is better than showing nothing, is a principle I embrace here.

Two things come to me from the mockups:

1. The mixed value in color looks like "white" color is selected. We might need to find a different symbol for showing next values, something similar or based on the `indeterminate` state of checkboxControl

<img width="120" alt="Image" src="https://github.com/user-attachments/assets/bbd119c5-ef2e-4708-8975-6ce88adac7d2" />

2. Not sure if it's something that you missed, but there is a font size control for the child blocks that, as per the current heuristic, is already controlled by the control in the Typography panel. The one in Typography should show a mixed value, and the one in Selection should show the value of each font size that exists in the selection.

| Selected blocks panel | Current block: Typography |
|---|---|
| <img width="1106" height="1032" alt="Image" src="https://github.com/user-attachments/assets/9058a20f-5469-49ce-95cd-50d3ea52cf3f" /> | <img width="1106" height="734" alt="Image" src="https://github.com/user-attachments/assets/e0d5fa88-4d00-4e09-bc43-20a163f2dc15" /> |

**Comment by jasmussen** [comment]
> similar or based on the indeterminate state of checkboxControl

Love it, yes, good instinct. 

> the one in Typography should show a mixed value, and the one in Selection should show the value of each font size that exists in the selection.

This relates to the more out-there Group mockup, yes? In that case, I agree it's not ideal that there are two typography controls, but just to clarify: the first one relates **only** to the Group itself, wheras the second one is meant to show mixed values of _all children_. It probably shouldn't be a font size control at all, maybe an itemgroup just like Colour. 

In fact, I tried a thing:

<img width="1560" height="2224" alt="Image" src="https://github.com/user-attachments/assets/ea95d063-066f-47f9-a1b6-416b6b6c04b5" />

I still don't think it's quite there, but hopefully it gets the idea across, that this panel—now Elements—is about the inner blocks, not the container itself.

**Comment by fcoveram** [comment]
> the first one relates only to the Group itself, whereas the second one is meant to show mixed values of all children. It probably shouldn't be a font size control at all, maybe an itemgroup just like Colour.

Then it's definitely not a control.

The value can not be changed from there but it works as an action to select the corresponding block. However, Elements (link, button, etc) can be changed from Elements panel, and therefore, each item is a control. With that in mind, elements in "Inner block styles" and "Element styles" should not look the same when they trigger different actions.

From the above, then my feedback is:
- Remove the `:` character at the end of the heading as it's not among the copy practices
- "Element style" controls
   - They should be placed on top as they are higher in the hierarchy due to the editing context.
   - Remove the target icon as they don't select the block but open the control
- "Inner block styles" actions
   - It's not clear whether the color item selects Typography or Background color. Unless this is intentional and the action selects all blocks that have that color set in both Typography and Background.
   - Color item shows the value (black or red), but Font Size and Radius do not. If there are two child blocks with different radii, which one is selected? 
- The target icon looks bolder. You have already noticed it, but adding it for the record.

**Comment by jasmussen** [comment]
Great feedback Fran, thank you! I'm still not sure about the panel, but I tried to address your feedback still to see if it took me in the right direction:

<img width="1560" height="2224" alt="Image" src="https://github.com/user-attachments/assets/b650d670-6e93-47b2-9edb-cfbbf714d09f" />

We're closing on on something I quite like here! The moving of Elements to the top also fixes an issue with this being a ToolsPanel where the ellipsis adds elements you can control there. By simply having block styles below, those will intrinsically be contextual to what a child does.

**Comment by fcoveram** [comment]
The above works well. Although I still believe that
* Maybe `blockDefault` icon works better than introducing a new one.
* Start simpler and not include multi-value styles such as radius, padding, margin, etc. In the case designed, the radius should show a specific value so all blocks with that value set are selected.

If we agree on the functionality, then when multiple blocks are selected:
- If value in control (ie, Size in Typograhpy) is…
   - Single (ie, Medium) = All blocks selected have the same value and the control modifies all to another single value.
   - Mixed = Blocks selected have different values and the control modifies all to have the same value.
- If value in selector (ie, Color in Element's Block Styles) is…
   - Single (ie, Red) = All blocks selected have the same value and the item selects all blocks with that value.
   - Mixed = **This one is contradictory.** The section lists all values of the selected block. Selecting a "Mixed" value is not clear about what will be selected.

The logic described above is what Figma does. There are no mixed items in the "Selection colors"

<img width="250" alt="Image" src="https://github.com/user-attachments/assets/c8423ca7-0054-4f8e-95a9-a757700799a3" />
